tests/benchmarks/multicore/idle_with_pwm/README.rst

```diff
@@ -0,0 +1,72 @@
+.. _multicore_idle_with_pwm_test:
+
+PWM in low power states test
+############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The test benchmarks the idle behavior of an application that runs on multiple cores and uses PWM driver to light up LED.
+
+Test scenario
+Repeat forever:
+- gradually light up LED (this lasts for ~1 second),
+- disable PWM (set period to 0); This will turn off LED,
+- sleep for 1 second.
```

Comment:
```suggestion
The test scenario repeats forever the following:

* Gradually light up the LED (this lasts for ~ one second).
* Disable the PWM (set period to 0) and turns off the LED.
* Sleep for one second.
```

Reply:
I'm not sure about
_Disable the PWM (set period to 0) and turns off the LED._
It's NOT two actions - set period to 0 then switch off LED.
Setting PWM period to 0 is a test step. As a result, LED switches OFF.
My intention was to explain that there are two observable "states":
a) LED lights up from OFF to "bright" which lasts for 1 sec;
b) LED is OFF for 1 sec.
In a) cpu is in power state active and it's using PWM driver.
In b) cpu is in low power state and PWM driver is disabled.

Reply:
What about
* Disable the PWM (set period to 0, which turns off the LED).
???

Reply:
I still don't like it.
You can light up / turn off LED with GPIO but this is abrupt change in brightness. Either 0%, either 100%.
Test focuses on enabling / disabling PWM driver. PWM allows to set any brightness between 0-100%.
LED is there only to indicate that PWM is indeed working.

Reply:
* Gradually increase PWM duty cycle from 0% to 50% in one second (the LED lights up), power state active.
  * Set PWM duty cycle to 0% for one second (PWM disabled, the LED is OFF), power state low.

Reply:
Well, then how about switching the order like this:
* PWM enabled, the LED lights up for one second (power state active).
* PWM disabled, the LED is turned off (power state low).

Or suggest something simple to outline what happens.

Reply:
We must have written at the same time. Okay with your suggestion, but *for* one second on the first bullet.

Reply:
In active state, every 20 ms, PWM duty cycle is increased by 1%. It is done 50 times, so it lasts for 50*20ms = 1 sec.
LED brightness increases from 0 to 50% (uhm, it's not correct; there is non-linear dependency between PWM duty cycle and brightness).
To enter sleep, PWM peripheral has to be disabled. There is no explicit API function. Driver disables PWM peripheral when duty cycle is set to 0% or 100%.